src/app/contact.service.ts

```diff
@@ -0,0 +1,37 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs/internal/BehaviorSubject';
+import { Observable } from 'rxjs/internal/Observable';
+import { map } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ContactService {
+  private _contacts: Contact[] = [];
+  private _contactSubject = new BehaviorSubject<Contact[]>([]);
```

Comment:
Nice to see you using `BehaviourSubject` - not strictly needed for this exercise, as just having an array would have worked fine - but it's great to see you using the RxJS library!